Remove unhighlighted comments

ulysseskan opened the pull request:
This implements most of the feature request in https://github.com/experimatt/hn-jobs-focus/issues/1 , but instead of adding it as an option, it removes unhighlighted comments by default.

Feel free to add it in the UI as a toggle or whatever you see fit.  Thanks for the cool little time-saver of an extension!

experimatt commented on scripts/content.js:
```diff
@@ -11,6 +11,7 @@ const getValuesFromStorage = async () => {
   chrome.storage.local.get(["highlightWords", "ignoreWords"]).then((result) => {
     values = result;
     applyHighlights(values);
+    removeUnhighlightedComments();    
```
Is this function call in the right place? There are two places where we're calling `applyHighlights`: here, and in the listener for when values are changed. Should this also be called in the listener as well?

ulysseskan replied:
Oops - thanks for catching this.  I added the missing function call in the listener.

experimatt reviewed (changes requested):
@ulysseskan thank you for throwing up a PR for this. I really appreciate it.

In addition to the comments I left, I'd also like to surface something (a radio button, checkbox, or similar) in the popup that a user can adjust based on their preferences. For example, choosing between highlighting ignored phrases or removing all un-highlighted comments entirely.

We can either go back and forth in this PR, or I can take your changes and run with them to implement what I'm describing. Let me know what you'd prefer - in either case, I may be slow to respond (or to make code changes).

But again, thank you for the suggestion and your work on this. I'm so glad you've found this extension to be helpful!

ulysseskan commented on scripts/content.js:
```diff
@@ -99,4 +100,10 @@ const clearAllHighlights = () => {
-}
\ No newline at end of file
+
+const removeUnhighlightedComments = () => {
```
Done.

ulysseskan commented:
Thanks for the review, Matt!  I added the check, fixed the missing function call, and added UI.

I'm also forcing a reload of the active tab upon ticking the box.  This resolves a problem I saw without the refresh - the page would be devoid of comments.

Not sure if it makes sense to also force a reload of the active tab when unticking the box?  Please feel free to let me know if you have any other suggestions or think things should be done differently.

experimatt commented:
@ulysseskan sorry it took me so long to get back to this. Having to reload the active tab was a smell to me, and I was able to figure out how to get it working like before, without any need for reloading.

I hope you don't mind, I took your changes, made some updates and adaptations, and pushed them to `main`. If you pull from `main` you should be able to see & use these updates.

Here's the commit/diff, for reference: https://github.com/experimatt/hn-jobs-focus/commit/c93a02ae4a64b87e705aa9e3b62918342b2d30a0

With that, I think we can close your PR. Thank you again for suggesting this change, and doing the work to update the code. My apologies again for the delay in reviewing this and following up!

ulysseskan commented:
@experimatt Finally got around to trying this out.  Very smooth and clean.  No reloads.  Much better than my version.  Thanks for running with the idea!
